### PR TITLE
More pbs compat fixes

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: src/sbatch/opt.c [hunk 1]
```diff
@@ -2472,14 +2472,19 @@ static void _parse_pbs_nodes_opts(char *node_opts)
 {
 	int i = 0;
 	char *temp = NULL;
-	int ppn = 0;
+	int tasks = 0;
+    int ppn = 0;
```
Comment (itkovian) [inline src/sbatch/opt.c]: indentation. something with tabs/spaces perhaps.

File: src/sbatch/opt.c [hunk 2]
```diff
@@ -2472,14 +2472,19 @@ static void _parse_pbs_nodes_opts(char *node_opts)
...
+	int max_ppn = 0;
 	int node_cnt = 0;
 	hostlist_t hl = hostlist_create(NULL);
 
 	while (node_opts[i]) {
 		if (!xstrncmp(node_opts+i, "ppn=", 4)) {
 			i+=4;
-			ppn += strtol(node_opts+i, NULL, 10);
+            ppn = strtol(node_opts+i, NULL, 10);
+			tasks += ppn;
```
Comment (itkovian) [inline src/sbatch/opt.c]: indentation.

File: src/sbatch/opt.c [hunk 3]
```diff
@@ -2472,14 +2472,19 @@ static void _parse_pbs_nodes_opts(char *node_opts)
...
+    int ppn = 0;
+	int max_ppn = 0;
 	int node_cnt = 0;
 	hostlist_t hl = hostlist_create(NULL);
 
 	while (node_opts[i]) {
 		if (!xstrncmp(node_opts+i, "ppn=", 4)) {
 			i+=4;
-			ppn += strtol(node_opts+i, NULL, 10);
+            ppn = strtol(node_opts+i, NULL, 10);
```
Comment (itkovian) [inline src/sbatch/opt.c]: indentation.

File: src/sbatch/opt.c [hunk 4]
```diff
@@ -2472,14 +2472,19 @@ static void _parse_pbs_nodes_opts(char *node_opts)
...
+			tasks += ppn;
+            if (ppn > max_ppn)
```
Comment (itkovian) [inline src/sbatch/opt.c]: indentation.

File: src/sbatch/opt.c [hunk 5]
```diff
@@ -2500,12 +2505,15 @@ static void _parse_pbs_nodes_opts(char *node_opts)
...
-	if (ppn) {
-		ppn *= node_cnt;
+	if (tasks) {
+		tasks *= node_cnt;
 		opt.ntasks_set = true;
-		opt.ntasks = ppn;
+		opt.ntasks = tasks;
 	}
 
+    if (max_ppn)
```
Comment (itkovian) [inline src/sbatch/opt.c]: indentation.

**Comment by stdweird** [comment]
@itkovian wrt the indentation, the file has mixed tabs / 4 spaces. i'm just adhering to their convertion :smile:

File: src/plugins/job_submit/pbs/spank_pbs.c [hunk 6]
```diff
@@ -61,19 +61,32 @@ int slurm_spank_task_init(spank_t sp, int ac, char **av)
...
 	if (getcwd(val, sizeof(val)))
 		spank_setenv(sp, "PBS_JOBDIR", val, 1);
 
-	if (spank_getenv(sp, "SLURM_JOB_ID", val, sizeof(val)) ==
-	    ESPANK_SUCCESS)
-		spank_setenv(sp, "PBS_JOBID", val, 1);
+
+	if (spank_getenv(sp, "SLURM_ARRAY_JOB_ID", val, sizeof(val)) ==
+	    ESPANK_SUCCESS) {
+        /* 20 is enough for unit32 (max index is 4M) */
```
Comment (itkovian) [inline src/plugins/job_submit/pbs/spank_pbs.c]: uint32

**Pull request merged**